pkg/windows/windows.go

```diff
@@ -602,17 +601,41 @@ func (vm *windows) ConfigureKubeProxy(nodeName, hostSubnet string) error {
 
 // Interface helper methods
 
-// ensureHostName ensures hostname of the Windows VM matches the expected name
-func (vm *windows) ensureHostName() error {
-	hostNameChangedNeeded, err := vm.isHostNameChangeNeeded()
+// ensureHostNameAndContainersFeature ensures hostname of the Windows VM matches the expected name
+// and the required Windows feature is enabled.
+func (vm *windows) ensureHostNameAndContainersFeature() error {
+	rebootNeeded := false
+	// Set the hostName of the Windows VM if needed
+	if vm.instance.NewHostname != "" {
+		hostNameChangedNeeded, err := vm.isHostNameChangeNeeded()
+		if err != nil {
+			return err
+		}
+		if hostNameChangedNeeded {
+			if err := vm.changeHostName(); err != nil {
+				return err
+			}
```

Comment:
could these two functions be combined and simplified into a single ensureHostName function?

Reply:
@sebsoto  sure seems doable, but I wanted to keep them together to figure out if the reboot is needed. Else I would have to make enusreHostName() return a bool to signal a reboot which seems a bit counter-intutive to ensure host name has been changed.